client.go

```diff
@@ -111,6 +111,10 @@ func NewClient(options Options) *Client {
 		backoff = options.Backoff
 	}
 
+	// add timeout to clients
+	httpclient.Timeout = options.Timeout
+	httpclient2.Timeout = options.Timeout
+
 	// if necessary adjusts per-request timeout proportionally to general timeout (30%)
 	if options.Timeout > time.Second*15 && options.RetryMax > 1 && !options.NoAdjustTimeout {
 		httpclient.Timeout = time.Duration(options.Timeout.Seconds()*0.3) * time.Second
```

Comment:
@Mzack9999  not sure about this (proportional timeout).

Ex: if target.com is slow to respond takes 20sec . If I change Timeout to 30sec . it is autoadjusted by default to approx 10 sec . hence client will fail after all 3 retries . 

If this is not imposed client will at least get 1 response

Reply:
Shall we set `NoAdjustTimeout` to `true` if the user provided a custom timeout?
In general, we lower timeout to circa 30% with many retries to speed up elaboration and consider we have linear/exponential backoff

Reply:
@Mzack9999 , yes setting `NoAdjustTimeout` to false by default would be enough for now . 
Not sure which OSI layer errors we are targeting with retry but assuming we are retrying network errors , how about moving this logic to lowerlevel  (dial timeout or tls timeout ??)

Reply:
Timeout/Retries are generically attempting to deal with both network/application level errors by ensuring we retry a consistent amount of times to overcome generic temporary errors and, at the same time, wait not too long for a failure to release resources. We can split them, but we would need more fine-grained settings, as it happens with the `http.Transport` layer.